### PR TITLE
[test] test the Thread network 1 hop throughput

### PR DESCRIPTION
**Pull request by zhanglongxia**
This commit adds a test case to the `cp-tests` to test the Thread network 1 hop throughput.

### CODE REVIEW

File: tools/cp-caps/README.md [hunk 1]
```diff
@@ -63,6 +63,7 @@ options:
   -c, --csl            test whether the RCP supports CSL transmitter
   -d, --diag-commands  test whether the RCP supports all diag commands
   -p, --data-poll      test whether the RCP supports data poll
+  -t, --throughput     test the Thread network 1 hop throughput
```
Comment (superwhd) [inline tools/cp-caps/README.md:66]:
```suggestion
  -t, --throughput     test the Thread network 1-hop throughput
```
nit

File: tools/cp-caps/rcp_caps_test.py [hunk 2]
```diff
@@ -80,17 +81,79 @@ def test_diag_commands(self):
...
         self.__test_data_poll_child()
 
+    def test_throughput(self):
+        """Test Thread network 1 hop throughput."""
+        if not self.__dut.support_iperf3():
+            print('The DUT doesn\'t support the tool "iperf3"')
+            return
+
+        if not self.__ref.support_iperf3():
+            print('The reference device doesn\'t support the tool "iperf3"')
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:101]: nit: probably you don't need to quote the `iperf3` so you don't have to escape the single quote.

Like:
```python
print("The reference device doesn't support the tool iperf3")
```

File: tools/cp-caps/rcp_caps_test.py [hunk 3]
```diff
@@ -80,17 +81,79 @@ def test_diag_commands(self):
...
+        ref_iperf3_server = threading.Thread(target=self.__ref_iperf3_server_task,
+                                             args=(ref_mleid, timeout),
+                                             daemon=True)
+        ref_iperf3_server.start()
+        self.__dut.wait(1)
+
+        results = self.__dut.iperf3_client(host=ref_mleid, bitrate=bitrate, transmit_time=transmit_time, length=length)
+        ref_iperf3_server.join()
+
+        if len(results) == 0:
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:133]: nit: maybe neater
```suggestion
        if not results:
```

File: tools/cp-caps/rcp_caps_test.py [hunk 4]
```diff
@@ -80,17 +81,79 @@ def test_diag_commands(self):
...
+        bitrate_string = ''
+
+        if bitrate >= 1000000:
+            bitrate_string = f'{round(bitrate/1000000, 3)} Mbits/sec'
+        elif bitrate >= 1000:
+            bitrate_string = f'{round(bitrate/1000, 3)} Kbits/sec'
+        else:
+            bitrate_string = f'{bitrate} bits/sec'
+
+        return bitrate_string
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:155]: Could be refactored like:
```

    units = ["bps", "Kbps", "Mbps", "Gbps", "Tbps"]
    unit_index = 0

    while bitrate >= 1000 and unit_index < len(units) - 1:
        bitrate /= 1000
        unit_index += 1

    return f"{bitrate:.2f} {units[unit_index]}"
```

**Review by superwhd — APPROVED** [review]
LGTM overall with minor suggestions

**Review by abtink — APPROVED** [review]
👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**